docs/dev/connectors/cassandra.md

```diff
@@ -78,77 +96,189 @@ Note that that enabling this feature will have an adverse impact on latency.
+DataStream<String> text = env.socketTextStream(hostname, port, "\n");
+
+// parse the data, group it, window it, and aggregate the counts
+DataStream<Tuple2<String, Long>> result = text
+        .flatMap(new FlatMapFunction<String, Tuple2<String, Long>>() {
+            @Override
+            public void flatMap(String value, Collector<Tuple2<String, Long>> out) {
+                // normalize and split the line
+                String[] words = value.toLowerCase().split("\\s");
+
+                // emit the pairs
+                for (String word : words) {
+                    //Do not accept empty word, since word is defined as primary key in C* table
+                    if (!word.isEmpty()) {
+                        out.collect(new Tuple2<String, Long>(word, 1L));
+                    }
+                }
+            }
+        })
+        .keyBy(0)
+        .timeWindow(Time.seconds(5))
+        .sum(1)
+        ;
+
+CassandraSink.addSink(result)
+        .setQuery("INSERT INTO example.wordcount(word, count) values (?, ?);")
+        .setHost("127.0.0.1")
+        .build();
 {% endhighlight %}
 </div>
+
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
-CassandraSink.addSink(input)
-  .setQuery("INSERT INTO example.values (id, counter) values (?, ?);")
-  .setClusterBuilder(new ClusterBuilder() {
-    override def buildCluster(builder: Cluster.Builder): Cluster = {
-      builder.addContactPoint("127.0.0.1").build()
-    }
-  })
+val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
+
+// get input data by connecting to the socket
+val text: DataStream[String] = env.socketTextStream(hostname, port, '\n')
+
+// parse the data, group it, window it, and aggregate the counts
+val result: DataStream[(String, Long)] = text
+  // split up the lines in pairs (2-tuples) containing: (word,1)
+  .flatMap(_.toLowerCase.split("\\s"))
+  .filter(_.nonEmpty)
+  .map((_, 1L))
+  // group by the tuple field "0" and sum up tuple field "1"
+  .keyBy(0)
+  .timeWindow(Time.seconds(5))
+  .sum(1)
+
+CassandraSink.addSink(result)
+  .setQuery("INSERT INTO example.wordcount(word, count) values (?, ?);")
+  .setHost("127.0.0.1")
   .build()
+
+result.print().setParallelism(1)
 {% endhighlight %}
 </div>
+
 </div>
 
-The Cassandra sinks support both tuples and POJO's that use DataStax annotations.
-Flink automatically detects which type of input is used.
 
-Example for such a Pojo:
+### Cassandra Sink Example for Streaming POJO Data Type
+An example of streaming a POJO data type and store the same POJO entity back to Cassandra. In addition, this POJO implementation needs to follow [DataStax Java Driver Manual](http://docs.datastax.com/en/developer/java-driver/2.1/manual/object_mapper/creating/) to annotate the class as Cassandra connector internally maps each field of this entity to an associated column of the desginated Table using `com.datastax.driver.mapping.Mapper` class of DataStax Java Driver.
```

Comment:
-> to annotate the class **as each field of this entity is mapped to an associated column of the designated table using the DataStax Java Driver `com.datastax.driver.mapping.Mapper` class.**